### PR TITLE
Feature/autodiff test

### PR DESCRIPTION
**Pull request by VIVelev**
Mainly test auto diff engine against PyTorch, others are tweaks.

### CODE REVIEW

File: tests/test_math/test_aggregate.py [hunk 1]
```diff
@@ -6,24 +6,33 @@
...
+    assert allclose(inputs[0].grad.value, (output / inputs[0]).value)
+
     assert (aggregate.prod(*inputs) == prod(inputs)).all()
 
 
 def test_mean(inputs):
-    assert allclose(aggregate.mean(inputs[0]).value, mean(inputs[0].value))
+    output = aggregate.mean(inputs[0])
+    assert allclose(output.value, mean(inputs[0].value))
+    assert (inputs[0].grad == 1 / 9).all()
```
Comment (VIVelev) [inline tests/test_math/test_aggregate.py]: Generalize `1/9`

**Review by VIVelev — COMMENTED** [review]
LGTM, @ManiacMaxo merge?

File: tests/test_math/test_aggregate.py [hunk 2]
```diff
@@ -4,26 +4,53 @@
 import nujo.math.aggregate as aggregate
 from nujo.init.random import rand
 
+# ====================================================================================================
+# Test Summation
+
 
 def test_sum(inputs):
-    assert (aggregate.sum(inputs[0]) == sum(inputs[0].value)).all()
+    output = aggregate.sum(inputs[0])
```
Comment (ManiacMaxo) [inline tests/test_math/test_aggregate.py:12]: Why make a new variable when you only use it once

Comment (VIVelev) [reply tests/test_math/test_aggregate.py:12]: Usually, I would not do that, you are right,  however here I did it for two reasons:
 1) Improve readability.
 2) Make it consistent with `test_prod` where `output` is used twice.

Comment (ManiacMaxo) [reply tests/test_math/test_aggregate.py:12]: I see, LGTM :)

**Pull request merged**